### PR TITLE
Fix reports of object-shorthand and prefer-rest-params

### PR DESCRIPTION
**Pull request by Josh-Cena**
#### Motivation

Rest parameters make the intent clearer; object shorthands are also more concise and widely adopted in production.

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/learn/server-side/express_nodejs/deployment/index.md [hunk 1]
```diff
@@ -115,21 +115,19 @@ For example, the code fragment below shows how you might set up "author" logging
 The debug variable is declared with the name 'author', and the prefix "author" will be automatically displayed for all logs from this object.
 
 ```js
-var debug = require('debug')('author');
+const debug = require('debug')('author');
```
Comment (Josh-Cena) [inline files/en-us/learn/server-side/express_nodejs/deployment/index.md:118]: Review tip: the express tutorials are re-formatted. View the diff with "hide whitespace" on.

File: files/en-us/learn/server-side/express_nodejs/forms/create_book_form/index.md [hunk 2]
```diff
@@ -51,72 +58,90 @@ Find the exported `book_create_post()` controller method and replace it with the
...
+        (err, results) => {
+          if (err) {
+            return next(err);
+          }
+
+          // Mark our selected genres as checked.
+          for (let i = 0; i < results.genres.length; i++) {
+            if (book.genre.indexOf(results.genres[i]._id) > -1) {
+              results.genres[i].checked = "true";
+            }
```
Comment (teoli2003) [inline files/en-us/learn/server-side/express_nodejs/forms/create_book_form/index.md:122]:
```suggestion
          for (const genre of results.genres) {
            if (book.genre.indexOf(genre._id) > -1) {
              genre.checked = "true";
            }
```

File: files/en-us/learn/server-side/express_nodejs/forms/create_book_form/index.md [hunk 3]
```diff
@@ -51,72 +58,90 @@ Find the exported `book_create_post()` controller method and replace it with the
...
+    } else {
+      // Data from form is valid. Save book.
+      book.save((err) => {
+        if (err) {
+          return next(err);
         }
+        //successful - redirect to new book record.
+        res.redirect(book.url);
+      });
     }
```
Comment (teoli2003) [inline files/en-us/learn/server-side/express_nodejs/forms/create_book_form/index.md:143]:
```suggestion
    }

    // Data from form is valid. Save book.
    book.save((err) => {
      if (err) {
        return next(err);
      }
      
      // Successful: redirect to new book record.
      res.redirect(book.url);
    });
```

File: files/en-us/learn/server-side/express_nodejs/forms/create_bookinstance_form/index.md [hunk 4]
```diff
@@ -51,46 +53,58 @@ Find the exported `bookinstance_create_post()` controller method and replace it
...
+    } else {
+      // Data from form is valid.
+      bookinstance.save((err) => {
+        if (err) {
+          return next(err);
         }
+        // Successful - redirect to new record.
+        res.redirect(bookinstance.url);
+      });
     }
```
Comment (teoli2003) [inline files/en-us/learn/server-side/express_nodejs/forms/create_bookinstance_form/index.md:106]:
```suggestion
    }

    // Data from form is valid.
    bookinstance.save((err) => {
      if (err) {
        return next(err);
      }
      
      // Successful: redirect to new record.
      res.redirect(bookinstance.url);
    });
```

File: files/en-us/learn/server-side/express_nodejs/forms/update_book_form/index.md [hunk 5]
```diff
@@ -66,74 +78,90 @@ Find the exported `book_update_post()` controller method, and replace it with th
...
+        (err, results) => {
+          if (err) {
+            return next(err);
+          }
+
+          // Mark our selected genres as checked.
+          for (let i = 0; i < results.genres.length; i++) {
+            if (book.genre.includes(results.genres[i]._id)) {
+              results.genres[i].checked = "true";
+            }
```
Comment (teoli2003) [inline files/en-us/learn/server-side/express_nodejs/forms/update_book_form/index.md:142]:
```suggestion
          for (const genre of results.genres) {
            if (book.genre.includes(genre._id)) {
              genre.checked = "true";
            }
```

File: files/en-us/learn/server-side/express_nodejs/forms/update_book_form/index.md [hunk 6]
```diff
@@ -66,74 +78,90 @@ Find the exported `book_update_post()` controller method, and replace it with th
...
+    } else {
+      // Data from form is valid. Update the record.
+      Book.findByIdAndUpdate(req.params.id, book, {}, (err, thebook) => {
+        if (err) {
+          return next(err);
         }
+        // Successful - redirect to book detail page.
+        res.redirect(thebook.url);
+      });
     }
```
Comment (teoli2003) [inline files/en-us/learn/server-side/express_nodejs/forms/update_book_form/index.md:163]:
```suggestion
    }

    // Data from form is valid. Update the record.
    Book.findByIdAndUpdate(req.params.id, book, {}, (err, thebook) => {
      if (err) {
        return next(err);
      }
      
      // Successful: redirect to book detail page.
      res.redirect(thebook.url);
    });
```

File: files/en-us/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md [hunk 7]
```diff
@@ -673,30 +673,29 @@ Let's write an example:
...
       driver.findElement(By.name('q')).sendKeys(webdriver.Key.TAB);
     });
 
     driver.findElement(By.name('btnK')).click();
 
-    driver.sleep(2000).then(function() {
-      driver.getTitle().then(function(title) {
+    driver.sleep(2000).then(() => {
+      driver.getTitle().then((title) => {
         if(title === 'webdriver - Google Search') {
```
Comment (teoli2003) [inline files/en-us/learn/tools_and_testing/cross_browser_testing/your_own_automation_environment/index.md:699]:
```suggestion
        if (title === 'webdriver - Google Search') {
```

File: files/en-us/mozilla/add-ons/webextensions/user_interface/notifications/index.md [hunk 8]
```diff
@@ -26,10 +26,10 @@ You then use {{WebExtAPIRef("notifications.create")}} to create your notificatio
 let title = browser.i18n.getMessage("notificationTitle");
 let content = browser.i18n.getMessage("notificationContent", message.url);
```
Comment (teoli2003) [inline files/en-us/mozilla/add-ons/webextensions/user_interface/notifications/index.md:27]:
```suggestion
const title = browser.i18n.getMessage("notificationTitle");
const content = browser.i18n.getMessage("notificationContent", message.url);
```

File: files/en-us/web/api/console/assert/index.md [hunk 9]
```diff
@@ -57,11 +57,11 @@ assertion:
...
-for (let number = 2; number <= 5; number += 1) {
-    console.log(`the # is ${number}`);
-    console.assert(number % 2 === 0, {number: number, errorMsg: errorMsg});
-    // or, using ES2015 object property shorthand:
-    // console.assert(number % 2 === 0, {number, errorMsg});
+for (let number = 2; number <= 5; number++) {
+  console.log(`the # is ${number}`);
+  console.assert(number % 2 === 0, { number, errorMsg });
+  // or, using ES2015 object property shorthand:
+  // console.assert(number % 2 === 0, {number, errorMsg});
```
Comment (teoli2003) [inline files/en-us/web/api/console/assert/index.md:64]: We do use it now by default…

```suggestion
```

File: files/en-us/web/api/usb/requestdevice/index.md [hunk 10]
```diff
@@ -59,16 +59,16 @@ be listed.
...
-});
+  { vendorId: 0x1209, productId: 0xa800 },
+  { vendorId: 0x1209, productId: 0xa850 },
+];
+navigator.usb.requestDevice({ filters })
+  .then((usbDevice) => {
+    console.log(`Product name: ${usbDevice.productName}`);
+  })
+  .catch((e) => {
+    console.log(`There is no device. ${e}`);
```
Comment (teoli2003) [inline files/en-us/web/api/usb/requestdevice/index.md:70]:
```suggestion
    console.error(`There is no device. ${e}`);
```

File: files/en-us/web/api/web_audio_api/migrating_from_webkitaudiocontext/index.md [hunk 11]
```diff
@@ -342,48 +342,46 @@ The `activeSourceCount` attribute has been removed from {{domxref("AudioContext"
...
-    const src = arguments[0];
-    const startArgs = Array.prototype.slice.call(arguments, 1);
-    src.onended = () => {
-      sources.splice(sources.indexOf(src), 1);
-    }
-    sources.push(src);
-    src.start.apply(src, startArgs);
+// Array to track the playing source nodes:
+const sources = [];
+// When starting the source, put it at the end of the array,
```
Comment (teoli2003) [inline files/en-us/web/api/web_audio_api/migrating_from_webkitaudiocontext/index.md:359]:
```suggestion

// When starting the source, put it at the end of the array,
```

**Review by teoli2003 — CHANGES_REQUESTED** [review]
The _rest parameters_ makes code much cleaner.

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**